aws/table_aws_s3_object_version.go

```diff
@@ -0,0 +1,198 @@
+	return &plugin.Table{
+		Name:        "aws_s3_object_version",
+		Description: "AWS S3 Object Version",
+		List: &plugin.ListConfig{
+			Hydrate: listS3ObjectVersions,
+			Tags:    map[string]string{"service": "s3", "action": "ListObjectVersions"},
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "bucket_name", Require: plugin.Required, CacheMatch: "exact"},
+				{Name: "prefix", Require: plugin.Optional},
+				{Name: "encoding_type", Require: plugin.Optional},
+				{Name: "delimiter", Require: plugin.Optional},
+				{Name: "version_id_marker", Require: plugin.Optional},
+			},
+		},
+		HydrateConfig: []plugin.HydrateConfig{
+			{
+				Func: getBucketLocationForObjects,
+				Tags: map[string]string{"service": "s3", "action": "GetBucketLocation"},
+			},
+		},
+		Columns: awsAccountColumns([]*plugin.Column{
+			{
+				Name:        "bucket_name",
+				Description: "The name of the container bucket of this object.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("bucket_name"),
+			},
+			{
+				Name:        "delimiter",
+				Description: "The delimiter grouping the included keys.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "encoding_type",
+				Description: "Encoding type used by Amazon S3 to encode object key names in the XML response.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "version_id_marker",
+				Description: "Marks the last version of the key returned in a truncated response.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "prefix",
+				Description: "Selects objects that start with the value supplied by this parameter.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "is_truncated",
+				Description: "A flag that indicates whether Amazon S3 returned all of the results that satisfied the search criteria.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "common_prefixes",
+				Description: "All of the keys rolled up into a common prefix count as a single return when calculating the number of returns.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "delete_markers",
+				Description: "Specifies caching behavior along the request/reply chain.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "version",
+				Description: "Container for version information.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Version.Key"),
+			},
+			{
+				Name:        "region",
+				Description: "The AWS Region in which the object is located.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getBucketLocationForObjects,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+type ObjectVersionDetails struct {
```

Comment:
Why we need a custom structure here?

Reply:
@bigdatasourav, in our process, we're iterating over an [inner-level property](https://github.com/turbot/steampipe-plugin-aws/pull/2070/files#diff-e2fd0c572171341a4210a91d7f8b0f822c3096acfa07acc8a8a0ed1ce4e4d6b9R173) of the response to refine the result set, even though there are several useful properties at the top level.

I haven't been able to devise an alternative that circumvents the use of a custom struct. Would you be able to suggest any different strategies that might allow us to avoid employing a custom structure?